frontend/src/components/dashboard/create/govsg/GovsgMessages.tsx

```diff
@@ -0,0 +1,212 @@
+import axios from 'axios'
+
+import { capitalize } from 'lodash'
+
+import { useEffect, useState } from 'react'
+
+import Moment from 'react-moment'
+
+import overrideStylesTitleBar from '../../campaigns/OverrideTitleBar.module.scss'
+
+import styles from './../../campaigns/Campaigns.module.scss'
+
+import NoMatchDashboardImg from 'assets/img/no-match-dashboard.png'
+import { Campaign } from 'classes'
+import { Pagination, TitleBar } from 'components/common'
+
+const ITEMS_PER_PAGE = 10
+
+interface GovsgMessage {
+  name: string
+  mobile: string
+  data: string
+  passcode: string
+  status: string
+  sent: Date
+  officer: string
+}
+
+interface GovsgMessagesProps {
+  campaignId: Campaign['id']
+}
+
+export const GovsgMessages = ({ campaignId }: GovsgMessagesProps) => {
+  const [govsgMessagesDisplayed, setGovsgMessagesDisplayed] = useState([])
+  const [selectedPage, setSelectedPage] = useState(0)
+  const [hasFetchedGovsgMessages, setHasFetchedGovsgMessages] = useState(false)
+  const [govsgMessageCount, setGovsgMessageCount] = useState(0)
+
+  const fetchGovsgMessages = async (selectedPage: number) => {
+    const options = {
+      offset: selectedPage * ITEMS_PER_PAGE,
+      limit: ITEMS_PER_PAGE,
+    }
+    const response = await axios.get(`/campaign/${campaignId}/govsg/messages`, {
+      params: {
+        ...options,
+      },
+    })
+    const { messages: govsgMessages, total_count: totalCount } = response.data
+    setGovsgMessageCount(totalCount)
+    setGovsgMessagesDisplayed(govsgMessages)
+    setSelectedPage(selectedPage)
+    if (!hasFetchedGovsgMessages) setHasFetchedGovsgMessages(true)
+  }
+
+  useEffect(() => {
+    void fetchGovsgMessages(selectedPage)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPage])
+
+  const handlePageChange = (index: number) => {
+    void fetchGovsgMessages(index)
+  }
+
+  const columns = [
+    {
+      name: 'Recipient name'.toUpperCase(),
```

Comment:
Does`.toUpperCase()` transform the column name only? If not, why don't we just type the name in caps?

Reply:
Sure, I can do that.